### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by romainmenke**

### CODE REVIEW

File: link/README.md [hunk 1]
```diff
@@ -34,17 +44,40 @@ import (
 	"github.com/romainmenke/pusher/link"
 )
 
-func main() {
+func HandlerFunc(handlerFunc http.HandlerFunc) http.HandlerFunc {
+	return newPushHandlerFunc(handlerFunc)
+}
```
Comment (fd) [inline link/README.md:49]: Try this:

```go
func Handler(handler http.Handler) http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {

		if !link.CanPush(w, r) {
			handler(w, r)
			return
		}
  		  
		var p = pusher{writer: w}
		handler.ServeHTTP(&p, r)
  		  
	}
}
```

File: link/example/main.go [hunk 2]
```diff
@@ -62,3 +62,39 @@ func APICall(w http.ResponseWriter, r *http.Request) {
...
+		var p pusher
+		p = pusher{writer: w}
+		handler(&p, r)
+
+	})
+}
+
+type pusher struct {
+	writer http.ResponseWriter
+}
```
Comment (fd) [inline link/example/main.go]: not a complete responsewriter

File: link/link.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package link
+
+import "net/http"
+
+func CanPush(w http.ResponseWriter, r *http.Request) bool {
```
Comment (fd) [inline link/link.go:24]: why not make tyis private and a part of `Push`

File: link/link.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		}
+
+		var p pusher
+		p = pusher{writer: w}
+		handler(&p, r)
+
+	})
+}
+
+type pusher struct {
```
Comment (fd) [inline link/link.go]: Not a complete response writer

**Review by fd — CHANGES_REQUESTED** [review]

File: link/example/main.go (hunk 2 above)
Comment (romainmenke) [reply link/example/main.go]: https://github.com/romainmenke/pusher/blob/dev/link/response_writer.go

File: link/link.go (hunk 4 above)
Comment (romainmenke) [reply link/link.go]: https://github.com/romainmenke/pusher/blob/dev/link/response_writer.go

File: link/link.go (hunk 3 above)
Comment (romainmenke) [reply link/link.go:24]: Because it has to be evaluated with the request, not the response.

File: link/README.md (hunk 1 above)
Comment (romainmenke) [reply link/README.md:49]: https://github.com/romainmenke/pusher/blob/dev/link/link.go#L9-L21

**Pull request merged**